src/service/document-info-impl.js

```diff
@@ -73,13 +78,31 @@ export class DocInfo {
           : sourceUrl;
     }
     const pageViewId = getPageViewId(ampdoc.win);
+
+    const doc = ampdoc.win.document;
+    if (doc.head) {
+      for (let n = doc.head.firstElementChild; n; n = n.nextElementSibling) {
+        if (n.tagName != 'LINK') {
+          continue;
+        }
+        const rel = n.getAttribute('rel');
+        const href = n.getAttribute('href');
```

Comment:
If you read `.href` instead of using `getAttribute` you are getting a full resolved URL and can skip the expensive `parseUrl` below.